pkg/provider/provider.go

```diff
@@ -93,8 +96,13 @@ func (p *KedaProvider) GetExternalMetric(namespace string, metricSelector labels
 				if err != nil {
 					logger.Error(err, "error getting metric for scaler", "ScaledObject.Namespace", scaledObject.Namespace, "ScaledObject.Name", scaledObject.Name, "Scaler", scaler)
 				} else {
+					for _, metric := range metrics {
+						metricValue, _ := metric.Value.AsInt64()
+						p.metricsExporter.RecordHPAScalerMetrics(namespace, metric.MetricName, metricSelector.String(), metricValue)
+					}
 					matchingMetrics = append(matchingMetrics, metrics...)
 				}
+				p.metricsExporter.RecordHPAScalerErrors(namespace, info.Metric, metricSelector.String(), err)
```

Comment:
It seems that if errors returned before scales loop then keda_hpa_scaler_error_totals does not increased (like in this [line](https://github.com/tbickford/keda/blob/3974023c90a477518ee1e6451e0357cb884ecbf8/pkg/provider/provider.go#L86)).
Is it supposed to be like this?

Reply:
That is a good point. The prometheus metric as it currently is expects the metric name as a label. If an error occurs in retrieving the scaler then there would be no metric so it may make sense to track errors by scaled object as well as by scaler.